admin/src/js/controllers/edit-user.js

```diff
@@ -305,11 +320,7 @@ angular
         .then(existingModel => {
           const updates = {};
           getUpdatedKeys(model, existingModel).forEach(key => {
```

Comment:
I think `getUpdatedKeys` needs to be changed as well, right now it will do a === comparison between role arrays, which will _always_ return false.